src/apps/project-home/ProjectHome.tsx

```diff
@@ -44,184 +66,249 @@ export const ProjectHome = (props: ProjectHomeProps) => {
+                <Plus size={20} /> <span>{t['Add Document']}</span>
+              </button>
+              <a
                 href={`/${lang}/projects/${project.id}/export/csv`}
-                className="button">
+                className='button'
+              >
                 <DownloadSimple size={20} />
                 <span>{t['Export annotations as CSV']}</span>
               </a>
             </div>
           )}
         </div>
 
-        <div 
-          className="project-home-grid-wrapper"
-          {...(canUpload ? getRootProps() : {})}>
-
-          <div className="project-home-grid" style={isDragActive ? { pointerEvents: 'none'} : undefined}>
-            {documents.map(document => (
+        <div
+          className='project-home-grid-wrapper'
+          {...(canUpload ? getRootProps() : {})}
+        >
+          <div
+            className='project-home-grid'
+            style={isDragActive ? { pointerEvents: 'none' } : undefined}
+          >
+            {documents.map((document) => (
               <DocumentCard
                 key={document.id}
                 isDefaultContext
                 isAdmin={isAdmin}
-                i18n={props.i18n} 
-                document={document} 
+                i18n={props.i18n}
+                document={document}
                 context={defaultContext!}
-                onDelete={() => onDeleteDocument(document)} 
+                onDelete={() => onDeleteDocument(document)}
                 onUpdate={onUpdateDocument}
                 onError={onError}
-                />
+              />
             ))}
           </div>
 
           {isDragActive && (
-            <div className="dropzone-hint-wrapper">
-              <div className="dropzone-hint">
-                <div className="dropzone-hint-popup">
+            <div className='dropzone-hint-wrapper'>
+              <div className='dropzone-hint'>
+                <div className='dropzone-hint-popup'>
                   <CloudArrowUp size={32} />
-                  <h1>Drop files or links to IIIF manifests to add them to your project.</h1>
-                  <p>
-                    Supported file formats: plaint text (UTF-8)
-                  </p>
+                  <h1>
+                    Drop files or links to IIIF manifests to add them to your
+                    project.
+                  </h1>
+                  <p>Supported file formats: plaint text (UTF-8)</p>
                 </div>
               </div>
             </div>
           )}
         </div>
-
-        <UploadTracker 
+        <div style={isDragActive ? { pointerEvents: 'none' } : undefined}>
+          <DocumentLibrary
+            open={addOpen}
+            i18n={props.i18n}
+            onAddDocument={onDocumentSelected}
+            onCancel={() => setAddOpen(false)}
+            user={props.user}
+            dataDirty={dataDirty || documentUpdated}
+            clearDirtyFlag={() => {
```

Comment:
Will the `dataDirty` flag trigger a full refetch of all documents in Supabase whenever the user makes a change? If so: might not be an issue immediately, but would we have to think about a lighter approach eventually? (In other cases, I updated state locally adding/replacing/deleting only single items.)